sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -845,6 +922,10 @@ object DateTimeUtils {
       format.toString.toUpperCase match {
         case "YEAR" | "YYYY" | "YY" => TRUNC_TO_YEAR
         case "MON" | "MONTH" | "MM" => TRUNC_TO_MONTH
+        case "DAY" | "DD" => TRUNC_TO_DAY
+        case "HOUR" | "HH" => TRUNC_TO_HOUR
+        case "MI" => TRUNC_TO_MINUTE
+        case "SEC" | "SS" => TRUNC_TO_SECOND
```

Comment:
These names were taken after https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions230.htm#i1002084 and http://www.ibm.com/support/knowledgecenter/SSGU8G_12.1.0/com.ibm.sqls.doc/ids_sqs_1481.htm.